alfresco-transform-imagemagick/alfresco-transform-imagemagick/src/main/java/org/alfresco/transformer/executors/ImageMagickCommandExecutor.java

```diff
@@ -76,6 +80,14 @@ protected RuntimeExec createTransformCommand()
         processProperties.put("MAGICK_HOME", ROOT);
         processProperties.put("DYLD_FALLBACK_LIBRARY_PATH", DYN);
         processProperties.put("LD_LIBRARY_PATH", DYN);
+
+        //Optional Windows properties
+        if ((CODERS != null && !CODERS.isBlank())
+                && (CONFIG != null && !CONFIG.isBlank()))
+        {
+            processProperties.put("MAGICK_CODER_MODULE_PATH", CODERS);
+            processProperties.put("MAGICK_CONFIGURE_PATH", CONFIG);
+        }
```

Comment:
Are these two properties always mutually dependent or can then be set independently ?

If the latter, then we should split into two separate if blocks ?

Reply:
I assumed all 5 properties are required (and all 5 are different than the linux ones) for Windows base on this repo props file. [link](https://github.com/Alfresco/alfresco-repository/blob/d1216de8d447280952f5a4c32885144a33ddbb89/src/main/resources/alfresco/subsystems/thirdparty/default/imagemagick-transform.properties). I couldn't find any relevant ImageMagick docs which prove that. I am currently trying to test that in Windows environment.

Reply:
Decided to split for simplicity and Alfresco [docs ](https://docs.alfresco.com/6.2/tasks/imagemagick-config.html)already specify what should be included for windows. Since all 5 props need to be different than default linux props, the permutations of user errors are too many to catch in this case.